v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstogcs/BigtableChangeStreamsToGcs.java

```diff
@@ -0,0 +1,149 @@
+        category = TemplateCategory.STREAMING,
+        displayName = "Cloud Bigtable change streams to Cloud Storage",
+        description =
+                "Streaming pipeline. Streams Bigtable change stream data records and writes them into a Cloud Storage bucket using Dataflow Runner V2.",
+        flexContainerName = "bigtable-changestreams-to-gcs",
+        contactInformation = "https://cloud.google.com/support",
+        optionsClass = BigtableChangeStreamsToGcsOptions.class)
+public class BigtableChangeStreamsToGcs {
+    private static final Logger LOG = LoggerFactory.getLogger(BigtableChangeStreamsToGcs.class);
+    private static final String USE_RUNNER_V2_EXPERIMENT = "use_runner_v2";
+
+    public static void main(String[] args) {
+        LOG.info("Starting Input Files to GCS");
+
+        BigtableChangeStreamsToGcsOptions options =
+                PipelineOptionsFactory.fromArgs(args).as(BigtableChangeStreamsToGcsOptions.class);
+
+        run(options);
+    }
+
+    private static String getProjectId(BigtableChangeStreamsToGcsOptions options) {
+        return options.getBigtableProjectId().isEmpty()
+                ? options.getProject()
+                : options.getBigtableProjectId();
+    }
+
+    public static PipelineResult run(BigtableChangeStreamsToGcsOptions options) {
+        LOG.info("Requested File Format is " + options.getOutputFileFormat());
+        options.setStreaming(true);
+        options.setEnableStreamingEngine(true);
+
+        final Pipeline pipeline = Pipeline.create(options);
+
+        // Get the Bigtable project, instance, database, and change stream parameters.
+        String projectId = getProjectId(options);
+        String instanceId = options.getBigtableInstanceId();
+
+        // Retrieve and parse the start / end timestamps.
+        Timestamp startTimestamp =
+                options.getStartTimestamp().isEmpty()
+                        ? Timestamp.now()
+                        : Timestamp.parseTimestamp(options.getStartTimestamp());
+        Timestamp endTimestamp =
+                options.getEndTimestamp().isEmpty()
+                        ? Timestamp.MAX_VALUE
+                        : Timestamp.parseTimestamp(options.getEndTimestamp());
+
+        // Add use_runner_v2 to the experiments option, since Change Streams connector is only supported
+        // on Dataflow runner v2.
+        List<String> experiments = options.getExperiments();
+        if (experiments == null) {
+            experiments = new ArrayList<>();
+        }
+        boolean hasUseRunnerV2 = false;
+        for (String experiment : experiments) {
+            if (experiment.equalsIgnoreCase(USE_RUNNER_V2_EXPERIMENT)) {
+                hasUseRunnerV2 = true;
+                break;
+            }
+        }
+        if (!hasUseRunnerV2) {
+            experiments.add(USE_RUNNER_V2_EXPERIMENT);
+        }
+        options.setExperiments(experiments);
+
+        String metadataTableName =
```

Comment:
I don't think 

options.getBigtableMetadataTableId() == null

can even happen because the default is empty string.  I am suggesting a different default.

Reply:
I changed the option to default to a string, this wont happen.
Removed this check.